python/numpydoc.rst

```diff
@@ -589,9 +589,15 @@ This works for both built-in types and most importable objects:
        [...]
 
 In general, provide the full namespace to the object, such as ```lsst.pipe.base.Struct```.
-It may be possible to reference objects in the same namespace as the current module without any namespace prefix.
+It may be possible to reference objects in the same package as the current type or function without any namespace prefix.
 Always check the compiled documentation site to ensure the link worked.
 
+.. note::
+
+   Relative links to types are evaluated relative to the *package* in which the linking object is documented, not the module file in which the docstring is written.
+   In particular, importing a type into a module does not let you use unqualified links to that type from module code.
```

Comment:
Sphinx gets type annotations correct in the signature these days even if it doesn't get the numpydoc part correct. I'm not sure where we should be mentioning type annotations.

How does something like a bare `DatasetType` work in numpydoc then if we say that importing into a module does not let you use it? lsst/daf_butler#846 does do that sometimes.

Reply:
Both [`Registry`](https://pipelines.lsst.io/v/daily/py-api/lsst.daf.butler.Registry.html) and [`DatasetType`](https://pipelines.lsst.io/v/daily/py-api/lsst.daf.butler.DatasetType.html) are documented in the same package, `lsst.daf.butler`. Their modules are irrelevant.